src/modules/commander/Commander.cpp

```diff
@@ -2613,6 +2622,31 @@ Commander::run()
 			}
 		}
 
+		// handle remote power off request
+		if (!armed.armed && last_power_off_sw_pos != sp_man.power_off_switch) {
```

Comment:
Needs to be coordinated with the state machine for shutdown. The shutdown logic shouldn't be specific to the RC switch.

Reply:
Added new power off state and corresponding state transition

Reply:
I'll review in detail later, but I was thinking it could probably be the same state if given an appropriate name. The point is that only the state machine should be deciding if it's safe to shutdown or reboot. There are other offenders in the system that need to be fixed.